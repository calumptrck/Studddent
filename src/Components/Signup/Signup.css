

.signup {
  margin-top: 70px;
  display:inline-block;
}

.mailtitle {
  font-family: Work Sans, sans-serif;
  font-weight: 600;
  font-size: 25px;
}

.maildesc {
  font-family: Work Sans;
  font-weight: 400;
  font-size: 16px;
  line-height: 22px;
}

.subscribe {
  background-color: rgb(243, 228, 93);
  margin-bottom: 20px;
  border-radius: 0.75em;
  font-size: 15px;
}

.subscribe:hover {
  background-color: rgb(255, 235, 57), 228, 93;
}

.response {
  display: none;
}

.hiddenInput {
  position: absolute; 
  left: -5000px;
}

.required {
  border-radius: 1em;
}

.grow2 { 
  transition: all .24s cubic-bezier(0,.5,.51,1); }
.grow2:hover { transform: scale(1.08); }

.shadow2 {
  box-shadow: 0 12px 25px rgba(0,0,0,0.04);
  -moz-box-shadow:    0 12px 25px rgba(0,0,0,0.04);
  -webkit-box-shadow: 0 12px 25px rgba(0,0,0,0.04);
}

.shadow2:hover {
  box-shadow: 0 14px 25px rgba(0,0,0,0.08);
  -moz-box-shadow:    0 14px 25px rgba(0,0,0,0.08);
  -webkit-box-shadow: 0 14px 25px rgba(0,0,0,0.08);
}

.clear {
  height: 50px;
  margin-top: -40px;
  margin-bottom: 80px;
}