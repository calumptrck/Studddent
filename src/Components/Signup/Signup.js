import React from 'react';
import './Signup.css'

import MoreButton from '../Buttons/MoreButton'

const Signup = ({ url, children }) =>
  <div className="signup">
    <h4 className="mailtitle mt-3">Interested in this project?</h4>
    <p className="maildesc mb-4">Sign up below to receive notifications for trending discounts, and updates on the project.</p>
    <div id="mc_embed_signup">
      <form action="https://calumpatrick.us3.list-manage.com/subscribe/post?u=b22b8149f5595dcf86cd74955&amp;id=0385a5d217" method="post"
        id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" className="validate" target="_blank" noValidate>
        <div id="mc_embed_signup_scroll">
          <div className="form-group">
            <label htmlFor="mce-EMAIL">Email Address</label>
            <input type="email" name="EMAIL" className="required form-control" id="mce-EMAIL" />
          </div>
          <div id="mce-responses" className="clear">
            <div className="response" id="mce-error-response"></div>
            <div className="response" id="mce-success-response"></div>
          </div>
          <div className="hiddenInput" aria-hidden="true">
            <input type="text" name="b_b22b8149f5595dcf86cd74955_0385a5d217" tabIndex="-1" value="" />
          </div>
          <div className="clear">
            <MoreButton type="submit">Subscribe</MoreButton>
          </div>
        </div>
      </form>
    </div>
  </div>

export default Signup