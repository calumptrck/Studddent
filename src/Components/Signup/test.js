import React from 'react';
import ReactDOM from 'react-dom';
import Signup from './Signup';
import renderer from 'react-test-renderer';

const tags = ['design', 'development', 'utility']

describe('Home', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Signup />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Signup />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});