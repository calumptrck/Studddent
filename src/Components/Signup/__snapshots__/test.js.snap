// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home has a valid snapshot 1`] = `
<div
  className="signup"
>
  <h4
    className="mailtitle mt-3"
  >
    Interested in this project?
  </h4>
  <p
    className="maildesc mb-4"
  >
    Sign up below to receive notifications for trending discounts, and updates on the project.
  </p>
  <div
    id="mc_embed_signup"
  >
    <form
      action="https://calumpatrick.us3.list-manage.com/subscribe/post?u=b22b8149f5595dcf86cd74955&id=0385a5d217"
      className="validate"
      id="mc-embedded-subscribe-form"
      method="post"
      name="mc-embedded-subscribe-form"
      noValidate={true}
      target="_blank"
    >
      <div
        id="mc_embed_signup_scroll"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="mce-EMAIL"
          >
            Email Address
          </label>
          <input
            className="required form-control"
            id="mce-EMAIL"
            name="EMAIL"
            type="email"
          />
        </div>
        <div
          className="clear"
          id="mce-responses"
        >
          <div
            className="response"
            id="mce-error-response"
          />
          <div
            className="response"
            id="mce-success-response"
          />
        </div>
        <div
          aria-hidden="true"
          className="hiddenInput"
        >
          <input
            name="b_b22b8149f5595dcf86cd74955_0385a5d217"
            tabIndex="-1"
            type="text"
            value=""
          />
        </div>
        <div
          className="clear"
        >
          <button
            className="getButton grow2 shadow2"
            onClick={undefined}
            type="submit"
          >
            Subscribe
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
